finetune/target_models/sequence_labeling.py

```diff
@@ -85,7 +86,40 @@ def _target_encoder(self):
             return SequenceMultiLabelingEncoder(pad_token=self.config.pad_token)
         return SequenceLabelingEncoder(pad_token=self.config.pad_token)
 
-
+def _context_same_cluster(a, b, thresholds=None):
+    thresholds = thresholds or {"lr": 0.5, "tb": 0.5} 
+    for ai in a:
+        atb = abs(ai["bottom"] - ai["top"])
+        alr = abs(ai["right"] - ai["left"])
+        for bi in b:
+            btb = abs(bi["bottom"] - bi["top"])
+            blr = abs(bi["right"] - bi["left"])
+            
+            ttb = max(btb, atb)
+            tlr = max(blr, alr)
+            if min(abs(ai["left"] - bi["right"]), abs(ai["right"] - bi["left"])) < thresholds["lr"] * tlr:
```

Comment:
are you trying to make sure overlap is at least 50%? right now this would also be true if there's no overlap but the shortest distance between tokens is within 50% of the length of the larger token.